bin/dotd-hoc

```diff
@@ -0,0 +1,155 @@
+  LOGGER.info("#{Time.new.strftime('%A, %B %d %Y')}: #{name} is DotD")
+  name
+end
+
+def should_i(question)
+  loop do
+    print "Should I #{question}? [#{underline 'Y'}es]/#{underline 'S'}kip/#{underline 'Q'}uit: "
+    response = gets.strip.downcase
+    if 'yes'.start_with?(response) # also catches blank/return ;)
+      LOGGER.info(question)
+      yield
+      return
+    elsif 'skip'.start_with?(response) || 'no'.start_with?(response)
+      return
+    elsif 'quit'.start_with?(response)
+      puts "Something weird going on? Take a look at: http://wiki.code.org/display/PROD/Daily+Deployment"
+      exit(-1)
+    else
+      puts "Sorry, I didn't understand that.\n\n"
+    end
+  end
+end
+
+def press_enter_to_continue
+  puts dim 'Press enter to continue...'
+  gets
+end
+
+def wait_for(thing)
+  puts "\nWait for #{thing}"
+  press_enter_to_continue
+end
+
+def open_url(url)
+  open_in_default_browser(url)
+  puts dim "Browser didn't open? go to:"
+  puts dim url
+end
+
+def open_in_default_browser(url)
+  # Based on http://stackoverflow.com/a/14053693/5000129
+  if RbConfig::CONFIG['host_os'] =~ /linux|bsd/
+    system "sensible-browser \"#{url}\""
+  else
+    system "open \"#{url}\""
+  end
+end
+
+def content_push(server_name, environment_name, user_name)
+  puts <<-EOS.unindent
+    Follow the prompts given by the content-push script.
+      If you need to do something fancier (or if the changed files do not
+      "look ok" when it asks you), log in to staging and hand-craft the commit.
+  EOS
+  success = system "ssh -t gateway.code.org ssh -t #{server_name} \"#{environment_name}/bin/content-push --name=\\\"#{user_name}\\\"\""
+
+  # Notice if the content push fails, and be helpful about fixing the problem.
+  unless success
+    puts "#{bold 'Uh-oh!'}  It looks like the #{server_name} content push failed!"
+    should_i "ssh to #{server_name} so you can fix the problem manually" do
+      puts <<-EOS.unindent
+
+      #{bold "Connecting to #{server_name}."} Type 'logout' when you're done.
+
+      #{dim 'd[ o_0 ]b'}
+
+      EOS
+      system "ssh -t gateway.code.org ssh #{server_name}"
+      puts <<-EOS.unindent
+
+      #{dim 'd[ o_0 ]b'}
+
+      #{bold 'Welcome back!'}
+
+      EOS
+      press_enter_to_continue
+    end
+  end
+end
+
+def open_pull_request(base, compare, title)
+  open_url "https://github.com/code-dot-org/code-dot-org/compare/#{base}...#{compare}?expand=1&title=#{CGI.escape title}"
+end
+
+puts <<-EOS.unindent
+
+  #{bold 'Welcome to Robo-Dev-of-the-Day, Hour of Code edition!'}
+
+    d[ o_0 ]b
+
+  #{bold 'Documentation'}
+
+    Dev-of-the-Day Checklist
+    #{dim 'http://wiki.code.org/display/PROD/Dev+of+the+Day+Checklist'}
+
+    Known Issues
+    #{dim 'http://wiki.code.org/display/PROD/Known+Issues'}
+
+EOS
+
+user_name = ask_for_name
+
+should_i "start a morning test run" do
+  # TODO: More detail on this process.
+  puts "TBD: SSH to staging-next machine, pull latest changes, run tests"
```

Comment:
Need more information here - what's the process for running tests on the staging-next machine?  Does that environment automatically rebuild on every commit, unless it's in the middle of a full test run?